### PR TITLE
[WIP] Updates to improve nameplate support

### PR DESCRIPTION
**Pull request by ls-**
There's many unit checks in the code that were written when there's no nameplate support in oUF which makes using certain elements either very difficult or entirely impossible.

### CODE REVIEW

**Comment by Wutname1** [comment]
I wanted to mention threat needs a bit of a revamp to handle enemy nameplates. As it stands now it works to show what friendly units have threat but not for you to know what enemy you have/don't have threat on.

**Comment by zorker** [comment]
@Wutname1 Do you mean sth like this should be part of oUF core?

https://github.com/zorker/rothui/blob/master/wow8.0/oUF_Simple/core/functions.lua#L89
https://github.com/zorker/rothui/blob/master/wow8.0/oUF_Simple/core/functions.lua#L255

Result: http://cdn-wow.mmoui.com/preview/pvw70142.jpg

A unitframe with green healthbar is sth I have threat on.

**Comment by ls-** [comment]
@Wutname1, there's the `.feedbackUnit` option, you should try using it. Set it to `"player"`, and the element will behave differently, similar to [how nameplates work in tank mode in the default UI](https://github.com/Gethe/wow-ui-source/blob/live/FrameXML/CompactUnitFrame.lua#L612).

File: ouf.lua [hunk 1]
```diff
@@ -736,7 +736,13 @@ function oUF:SpawnNamePlates(namePrefix, nameplateCallback, nameplateCVars)
...
 		elseif(event == 'PLAYER_TARGET_CHANGED') then
 			local nameplate = C_NamePlate.GetNamePlateForUnit('target')
-			if(not nameplate) then return end
+			if(not nameplate) then
+				if(nameplateCallback) then
+					nameplateCallback(nil, event, 'target')
+				end
+
+				return
+			end
```
Comment (p3lim) [inline ouf.lua]: Documentation needs to be updated for this.

**Comment by p3lim** [comment]
Other than that it looks good to merge.

**Pull request merged**